strax/config.py

```diff
@@ -203,6 +218,224 @@ def validate(self, config,
             config[self.name] = self.get_default(run_id, run_defaults)
 
 
+#Backward compatibility
+@export
+class Config(Option):
+    def __init__(self, **kwargs):
+        if 'name' not in kwargs:
+            kwargs['name'] = ''
+        super().__init__(**kwargs)
+
+    def __set_name__(self, owner, name):
+        self.name = name
+        takes_config = {name: self}
+        if (hasattr(owner, 'takes_config')
+                and len(owner.takes_config)):
+            # Already have some options set, e.g. because of subclassing
+            # where both child and parent have a takes_config decorator
+            
+            if name in owner.takes_config:
+                raise RuntimeError(
+                    f"Attempt to specify option {name} twice")
+            owner.takes_config = immutabledict({
+                **owner.takes_config, **takes_config})
+        else:
+            owner.takes_config = immutabledict(takes_config)
+
+    def __get__(self, obj, objtype=None):
+        return self.fetch(obj)
+
+    def __set__(self, obj, value):
+        obj.config[self.name] = value
+
+    def fetch(self, plugin):
+        ''' This function is called when the attribute is being 
+        accessed. Should be overriden by subclasses to customize behavior.
+        '''
+        if hasattr(plugin, 'config') and self.name in plugin.config:
+            return plugin.config[self.name]
+        raise AttributeError('Plugin has not been configured.')
+
+@export
+class LookupConfig(Config):
+    mapping: ty.Mapping
+    keys = ty.Iterable
+
+    def __init__(self, mapping: ty.Mapping, keys=('name', 'value'), **kwargs):
+        super().__init__(**kwargs)
+        self.mapping = mapping
+        if not isinstance(keys, ty.Iterable):
+            keys = (keys,)
+        self.keys = keys
+        
+    def fetch(self, plugin):
+        key = []
+        for k in self.keys:
+            if k=='name':
+                v = self.name
+            elif k=='value':
+                v = plugin.config[self.name]
+            elif isinstance(k, str) and hasattr(plugin, k):
+                v = getattr(plugin, k)
+            else:
+                v = k
```

Comment:
at this point you lost me, maybe reviewing at this time also isn't the greatest idea but this is supposed to just return the key itself as the value?

Reply:
haha yeah i probably also wrote that sometime during the night... it allows for "constant" keys. So if I have a dict-like object where one of the keys needs to just always be some constant then I can pass that constant in the keys list and it will always  be included in the lookup key...